Compilation error on FreeBSD systems (0.30-TRIAL)

Compilation fails on FreeBSD systems:

```
 xs/libgit2/include/git2/transport.h:15:21: error: libssh2.h: No such file or directory
```

libssh2 is installed and the check_lib check before was apparently successful. Problem is that 3rd party libraries are install to /usr/local on *BSD systems, and probably Makefile.PL does not address this.

Regards, Slaven

--- a/inc/MakeMaker.pm
+++ b/inc/MakeMaker.pm
@@ -13,19 +13,46 @@ use Devel::CheckLib;
 
 my $def = '';
 my $lib = '';
+my $inc = '';
+
+my %os_specific = (
+	'freebsd' => {
+		'ssh2' => {
+			'inc' => ['/usr/local/include'],
+			'lib' => ['/usr/local/lib']
+		}
+	}
+);
 
 if (check_lib(lib => 'ssl')) {
 	$def .= ' -DGIT_SSL';
 	$lib .= ' -lssl -lcrypto';
 
 	print "SSL support enabled\n";
+} else {
+	print "SSL support disabled\n";
 }
 
 if (check_lib(lib => 'ssh2')) {
+	my $os = $^O;
+
+	if (my $os_params = $os_specific{$os}) {
+		if (my $ssh2 = $os_params{'ssh2'}) {
+			if (my $ssh2inc = $ssh2{'inc'}) {
+				$inc .= ' -I'.join (' -I', @$ssh2inc);
+			}
+			if (my $ssh2lib = $ssh2{'lib'}) {
+				$lib .= ' -L'.join (' -L', @$ssh2lib);
+			}
+		}
+	}
+
 	$def .= ' -DGIT_SSH';
 	$lib .= ' -lssh2';
 
 	print "SSH support enabled\n";
+} else {
+	print "SSH support disabled\n";
 }
 
 sub MY::c_o {
@@ -46,6 +73,7 @@ my {{ $WriteMakefileArgs }}
 
 $WriteMakefileArgs{DEFINE} .= $def;
 $WriteMakefileArgs{LIBS}   .= $lib;
+$WriteMakefileArgs{INC}    .= $inc;
 
 unless (eval { ExtUtils::MakeMaker->VERSION(6.56) }) {
 	my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};

